src/libprojectM/libprojectM_shared.cmake

```diff
@@ -71,7 +71,10 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
             )
 endif()
 
-enable_target_linker_flags_if_supported(projectM_shared PRIVATE --no-undefined)
+# --no-undefined doesn't make sense for Windows as all symbols are always required to be defined
+if (NOT WIN32)
```

Comment:
Better use `MSVC` instead of `WIN32`, as MinGW also sets WIN32 to `true` while using GCC, where the flag still makes sense.

Reply:
I make that change based on threads [here](https://github.com/mesonbuild/meson/issues/3670) and [here](https://github.com/mesonbuild/meson/pull/4230). Apparently, it has no effect for Windows binaries, as there can be no undefined symbols in DLLs when building.

MinGW GCC silently swallows the flag but does nothing with it, and according to the issue thread, the LLVM linker warns about ignoring the flag. Unfortunately I wasn't able to test that particular behaviour, not with MSVC LLD or MinGW LLD.

That being said, I can change it if you want.

Reply:
Ping, is this alright?

Reply:
Sure, as long as it compiles, it's fine!